File: CustomListView/app/src/main/java/com/example/customlistview/MainActivity.java
package com.example.customlistview;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    ListView a;
    List<item> list = new ArrayList<>();
    Adapter adapter;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        a = findViewById(R.id.id1);
        list.add(new item(R.drawable.neu,"Clash of Clan đang trực tiếp một game ","4 giờ trước"));
        list.add(new item(R.drawable.clan,"NEU conffesion vừa cập nhật một trạng thái của họ","1 giờ trước"));
        list.add(new item(R.drawable.chuot,"Minh Nguyễn đã đăng trong Bk đại cương môn phái","2 giờ trước"));
        list.add(new item(R.drawable.tri,"Huy Phu đã bày tỏ cảm xúc về bình luận của bạn","15 phút trước"));
        adapter = new Adapter(list,getApplicationContext());
        a.setAdapter(adapter);
    }
}

File: AsyncTask/app/src/main/java/com/example/asynctask/MainActivity.kt
package com.example.asynctask

import android.os.AsyncTask
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Adapter
import android.widget.ArrayAdapter
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*
import org.json.JSONArray
import org.json.JSONObject
import java.io.BufferedReader
import java.io.InputStream
import java.io.InputStreamReader
import java.lang.StringBuilder
import java.net.HttpURLConnection
import java.net.URL
import java.net.URLConnection

class MainActivity : AppCompatActivity() {
    var arrayList = ArrayList<String>()
    lateinit var adapterArray:ArrayAdapter<String>
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        adapterArray = ArrayAdapter(this,android.R.layout.simple_expandable_list_item_1,arrayList)
        listServer.adapter = adapterArray as ArrayAdapter<String>
        ReadContentUrl().execute("http://192.168.1.50/webservice/getData.php")
    }

    inner class ReadContentUrl : AsyncTask<String, Void, String>() {
        override fun onPreExecute() {
            super.onPreExecute()
        }
        override fun doInBackground(vararg params: String?): String {
            var content: StringBuilder = StringBuilder()
            val url: URL = URL(params[0])
            val urlConnection: HttpURLConnection = url.openConnection() as HttpURLConnection
            val inputStream: InputStream = urlConnection.inputStream
            val inputStreamReader: InputStreamReader = InputStreamReader(inputStream)
            val bufferedReader: BufferedReader = BufferedReader(inputStreamReader)
            var line: String = ""
            try {
                do {
                    line = bufferedReader.readLine()
                    if(line!=null){
                        content.append(line)
                    }
                } while (line != null)
                bufferedReader.close()
            } catch (e: Exception) {
                Log.d("Tri", e.toString())
            }
            return content.toString()
        }
        override fun onPostExecute(result: String?) {
            super.onPostExecute(result)
            var jsonArray:JSONArray = JSONArray(result)
            for(khoahoc in 0..jsonArray.length()-1){
                var objectKH = jsonArray.getJSONObject(khoahoc)
                var ten = objectKH.getString("TenKH")
                var hphi = objectKH.getString("HocPhi")
                arrayList.add(ten+" - "+hphi)
            }
            adapterArray.notifyDataSetChanged()
        }
    }
}
